Dockerfile

```diff
@@ -1,13 +1,22 @@
-FROM --platform=${BUILDPLATFORM:-linux/amd64} cgr.dev/chainguard/go:1.20 AS builder
+#syntax=docker/dockerfile:1.4
+
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.1.0 AS xx
+
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS builder
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
 ARG TARGETOS
 ARG TARGETARCH
 
+COPY --from=xx / /
 WORKDIR /app
 COPY . .
-RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -o pizza-oven .
+RUN --mount=target=/go/pkg/mod,type=cache \
+    --mount=target=/root/.cache,type=cache \
+    xx-go build -ldflags="${GO_LDFLAGS}" -o pizza-oven .
 
-FROM --platform=${TARGETPLATFORM:-linux/amd64} cgr.dev/chainguard/glibc-dynamic
+# FROM gcr.io/distroless/static-debian11
+# not sure if scratch would work (ssl needed maybe?)
```

Comment:
Very good question.

The `pizza-oven` server doesn't need SSL right now and in the future, I'd imagine we'd want TLS to terminate before it hits this service.

The argument could be made for cross container/pod TLS if this was being deployed within an untressed kubernetes network. But that's mostly congegture.

`scratch` is good for now. And in the future, we'd want to think deeply about the in-transit security.

```diff
- # FROM gcr.io/distroless/static-debian11
- # not sure if scratch would work (ssl needed maybe?)
```

Reply:
> chainguard images are only linux/amd64 and linux/arm64. so if we want to add darwin/arm64 , linux/armv7 (32bit), we need to consider other base image. I tested with scratch for now but I dont know pizza's requirements (ssl needed?) so maybe someone can chime in here?

Also see #11 : We won't be able to consume specific image tags beyond `latest` or `latest-dev`. In the future, OpenSauced may go that direction, but for now, to keep us going fast, using scratch is great.

So when this merges, it'll close #11